rellic/AST/Z3ConvVisitor.cpp

```diff
@@ -532,12 +532,20 @@ bool Z3ConvVisitor::VisitBinaryOperator(clang::BinaryOperator *c_op) {
   auto rhs = GetOrCreateZ3Expr(c_op->getRHS());
   // Create z3 binary op
   switch (c_op->getOpcode()) {
-    case clang::BO_LAnd:
-      InsertZ3Expr(c_op, Z3BoolCast(lhs) && Z3BoolCast(rhs));
+    case clang::BO_GT:
```

Comment:
We should differentiate between signed and unsigned comparisons here. If `lhs` of the comparison is signed, we should use, for example, `z3::sgt()`, otherwise `z3::ugt()`.